woocommerce/woocommerce
# Product Importer: Ensure all products in the file get processed

### All Submissions:


### Changes proposed in this Pull Request:

Changes the "percent complete" calculation in the product importer so that it never rounds up to 100% when some small percentage of the file hasn't actually been imported yet, causing some data to get skipped and not imported.

Closes #36618

### How to test the changes in this Pull Request:

1. Download the test CSV file [here](https://gist.github.com/coreymckrill/ed2ba276bc97e4b8b97bb889c112b766). (The sample.csv file in this changeset won't work because the Ajax-powered importer works in batches of 30 rows, so the number of rows needs to be a multiple of 30 + 1.)
2. Note that it has **211** data rows in it (not including the headings row).
3. Before switching to this branch, try importing the file:
    * Go to All Products > Import
    * Check the box to update existing products (this way the data rows won't actually be imported into your test site)
    * Continue and run the import
4. You should see that **210** products were skipped, despite there being 211 products in the file.
5. Now check out this branch and go through the same process. This time it should correctly show that 211 products were skipped.

### Other information:


### FOR PR REVIEWER ONLY:
